### DISCUSSION TITLE
Go 1.21

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#3109] [Q&A]
Go 1.21 is out and seems to bring some performance improvements in both compilation and runtime. @ganigeorgiev have you been doing any experimentation with it on PB at all?

I also noticed the new `slog` package, and came across this benchmark https://github.com/rs/zerolog/issues/571 thought it would be pretty interesting for you to know.

Keep up the great work 💪

**Comment by ganigeorgiev** [comment]
There are plans to bump the min required Go version to at least 1.20 in the future (_there are several todos throughout the code with various notes, mostly related to the multiple wrapped errors handling_) but the changes so far in the language are not that drastic to require it.

If you use PocketBase as Go framework, you can update your Go installation and PocketBase should compile just fine with it. Also note that the claimed runtime performance gains are mostly when the Profile-guided optimization is enabled.

**Reply by pnmcosta** [reply]
Thank you for the update. Yeah I'm going to read a bit more on Profile-guided optimization (it's preview anyway I think) and give it a go, I'll report back anything I may find.
